nucypher/datastore/db/models.py

```diff
@@ -14,18 +14,21 @@
 You should have received a copy of the GNU Affero General Public License
 along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
 """
+import msgpack
 from collections.abc import Mapping
 from datetime import datetime
 from sqlalchemy import (Boolean, Column, DateTime, ForeignKey, Integer, LargeBinary)
 from sqlalchemy.orm import relationship
-from typing import Callable, NamedTuple
+from typing import Any, Callable, NamedTuple, Optional, Union
 
 from nucypher.crypto.utils import fingerprint_from_key
 from nucypher.datastore.db import Base
 
 
-FieldEncoder = NamedTuple('FieldEncoder', [('encode', Callable),
-                                           ('decode', Callable)])
+class RecordField(NamedTuple):
+    field_type: Any
+    encode: Optional[Callable] = lambda field: field
+    decode: Optional[Callable] = lambda field: field
```

Comment:
Why set a default at all if it's an *empty callable*?

Reply:
This just simply allows us to call `encode` or `decode` regardless if the type needs to be encoded or decoded. It makes logic in `DatastoreRecord` much cleaner and avoids unnecessary “look before you leap” patterns.